js/eventloop/eventloop.go

```diff
@@ -55,11 +55,69 @@ func (e *EventLoop) wakeup() {
 	}
 }
 
-// RegisterCallback register that a callback will be invoked on the loop, preventing it from returning/finishing.
-// The returned function, upon invocation, will queue its argument and wakeup the loop if needed.
-// If the eventLoop has since stopped, it will not be executed.
-// This function *must* be called from within running on the event loop, but its result can be called from anywhere.
-func (e *EventLoop) RegisterCallback() func(func() error) {
+// RegisterCallback is deprecated, see ReservePendingCallback.
+//
+// Deprecated: due to the confusing name, we renamed this method to
+// ReservePendingCallback.
+func (e *EventLoop) RegisterCallback() func(func() error) { // TODO: remove
+	return e.ReservePendingCallback()
+}
+
+// ReservePendingCallback signals to the event loop that you are going to do
```

Comment:
I have some problems with the word `pending` as to me it means that there already is a callback but it's waiting on something. 

I kind of prefer `future` although [future](https://en.wikipedia.org/wiki/Futures_and_promises) also means a "reverse" promise so ... uh not certain it's all that better.

From googling on synonyms of future i get:
- later - ReserveLaterCallback :grimacing: 
- skipping a bunch of terrible ones
- upcoming - ReserveUpcomingCallback seems a bit better
- planned - ReservePlannedCallback - seems even better to me ... I even can see how `PlanACallback` can be a good name
- prospective - not really 
- expected - ReserveExpectedCallback is not the worse, but definitely not my favorite.

Reply:
For me, pending is the word that describes exactly what is going on here, as I said in [this comment](https://github.com/grafana/k6/pull/2615#discussion_r928728759). Its definitions are "about to happen or waiting to happen", "used to say that one thing must wait until another thing happens", "waiting to happen", "while waiting for something to happen or be decided", exactly what we're trying to communicate here, right?

Reply:
Adding a bit of entropy: `ReserveScheduledCallback`?